lib/crawler/keyword/helpers/csv_parser.ex

```diff
@@ -0,0 +1,51 @@
+defmodule Crawler.Keyword.Helpers.CSVParser do
+  alias NimbleCSV.RFC4180, as: CSV
+
+  @keywords_file_max_length 1000
+  @keyword_min_length 1
+  @keyword_max_length 100
+
+  def keywords_limit, do: @keywords_file_max_length
+  def keyword_min_length, do: @keyword_min_length
+  def keyword_max_length, do: @keyword_max_length
+
+  def parse(file_path) do
+    keywords = parse_keyword_list_from_file(file_path)
+
+    case validate_keywords_length(keywords) do
+      {:ok, keywords} -> validate_keyword_length(keywords)
+      {:error, reason} -> {:error, reason}
+    end
+  end
+
+  defp parse_keyword_list_from_file(path) do
+    path
+    |> File.stream!()
+    |> CSV.parse_stream(skip_headers: false)
+    |> Enum.to_list()
+    |> List.flatten()
+  end
+
+  defp validate_keyword_length(keywords) do
+    keyword_length_exceeded_limit_fn = fn keyword ->
```

Comment:
Is it `keywords` @vnntsu 

```suggestion
    keyword_length_exceeded_limit_fn = fn keywords ->
```

Reply:
It's `keyword` since we're getting one by one to check its length